zarr/_storage/v3_storage_transformers.py

```diff
@@ -32,7 +32,8 @@ class _ShardIndex(NamedTuple):
 
     def __localize_chunk__(self, chunk: Tuple[int, ...]) -> Tuple[int, ...]:
         return tuple(
-            chunk_i % shard_i for chunk_i, shard_i in zip(chunk, self.store.chunks_per_shard)
+            chunk_i % shard_i
+            for chunk_i, shard_i in zip(chunk, self.store.chunks_per_shard, strict=False)
```

Comment:
why is `strict=False`? Are the lengths of the two iterables fixed somewhere else?

Reply:
(ditto for all the other `zip` invocations)

Reply:
Admittedly I just blindly applied this change that `ruff` was asking for 🙈 . For simplicitly I've reverted the `strict=False` changes, and ignored the rule for now.

Reply:
is it this rule? https://docs.astral.sh/ruff/rules/zip-without-explicit-strict/ I think we can just comply by using `strict=True` in that case

Reply:
to clarify, I don't think there's any usage of zip in the codebase where we want the `strict=False` behavior, so `strict=True` should be fine.

Reply:
I thought for simpliclty in this PR it would be better to ignore the rule - happy to add `strict=True` everywhere instead if you think that's a better option?

Reply:
i think we should follow the rule and, for each invocation of `zip`, we decide if the usage is consistent with `strict=True` or `strict=False`. I'm pretty sure that we always use zip with the assumption that the two iterables have the same number of elements, so any test failures after `strict=False` would probably be real bugs.

Reply:
Do you think this PR should be blocked on that - I'm not sure I have the expertise to go through and decide each case, but I'm happy for someone else to push to this branch.

Reply:
no need to block, let's do it in a follow-up PR

Reply:
Fab - I can open an issue in a bit so this doesn't get lost